src/clearskies/backends/graphql_backend.py

```diff
@@ -0,0 +1,916 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable
+
+from clearskies import configs, configurable, decorators, loggable
+from clearskies.autodoc.schema import Integer as AutoDocInteger
+from clearskies.autodoc.schema import Schema as AutoDocSchema
+from clearskies.autodoc.schema import String as AutoDocString
+from clearskies.backends.backend import Backend
+from clearskies.clients import graphql_client as client
+from clearskies.di import InjectableProperties, inject
+from clearskies.functional import string as string_helpers
+
+if TYPE_CHECKING:
+    from clearskies import Column, Model
+    from clearskies.query import Query
+
+
+class GraphqlBackend(Backend, configurable.Configurable, InjectableProperties, loggable.Loggable):
+    """
+    Autonomous backend for integrating clearskies models with GraphQL APIs.
+
+    Dynamically constructs GraphQL queries by introspecting the clearskies Model.
+    Supports CRUD operations, pagination, filtering, and relationships.
+
+    Configuration:
+    - graphql_client: GraphqlClient instance (required)
+    - root_field: Override the root field name (optional, defaults to model.destination_name())
+    - pagination_style: "cursor" for Relay-style or "offset" for limit/offset (default: "cursor")
+    - api_case: Case convention used by the GraphQL API (default: "camelCase")
+    - model_case: Case convention used by clearskies models (default: "snake_case")
+    - is_collection: Explicitly set if this resource is a collection (True) or singular (False/None=auto-detect)
+    - include_relationships: Enable automatic relationship fetching (default: False for performance)
+    - max_relationship_depth: Maximum depth for nested relationships (default: 2)
+    - nested_relationships: Allow relationships within relationships (default: False)
+    - relationship_limit: Default limit for HasMany/ManyToMany relationships (default: 10)
+    - use_connection_for_relationships: Use connection pattern for collections (default: True)
+    """
+
+    # Tell clearskies that count() may not be reliable for all GraphQL APIs
+    # This prevents clearskies from trying to call count() in situations where
+    # it would fail (e.g., for relationship queries with incompatible filters)
+    can_count = True
+
+    # GraphQL connection (required)
```

Comment:
For any comment that is meant to be part of the documentation (and every property that corresponds to an argument to the `__init__` should be documented), you want the comment for that property to be set between a docblock-like string (e.g. triple apostrophes).  For the client in particular I would provide even just a basic example of how to construct it:

```
    """
    The graphql connection
    
    An instance of clearskies.clients.GraphqlClient.  For example:
    
    ```
    class Model
        backend = clearskies.backends.GraphqlBackend(
            graphql_client = clearskies.clients.GraphqlClient(endpoint="http://localhost:4000/graphql", authentication=clearskies.authentication.SecretBearer(environment_key="MY_AUTH_SECRET")
        )
    ```
    """
```